examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+expectation values.
+
+Putting these two results together, `Ryan Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__
+show that samples of the expectation value fed into the parameter shift rule provide
+unbiased estimators of the quantum gradient --- resulting in a form of stochastic gradient descent.
+Moreover, they show that convergence of the stochastic gradient descent is guaranteed
+to be on par with standard gradient descent, even in the surprising case where the number
+of shots is 1!
+
+.. note::
+
+    It is worth noting that the smaller the number of shots used, the larger the
+    variance in the estimated expectation value. As a result, it may take
+    more optimization steps for convergence than using a larger number of shots,
+    or the analytic gradient.
+
+    At the same time, a reduced number of shots may significantly reduce the
+    wall time of each optimization step, leading to a reduction in the overall
+    optimization time.
+
+Let's consider a simple example in PennyLane, comparing analytic gradient
+descent (with exact expectation values) to stochastic gradient descent
+using a finite number of shots.
+
+"""
+
+##############################################################################
+# A single shot stochastic gradient descent
+# -----------------------------------------
+#
+# Consider the Hamiltonian
+#
+# .. math::
+#
+#     H = \begin{bmatrix}
+#           8 & 4 & 0 & -6\\
+#           4 & 0 & 4 & 0\\
+#           0 & 4 & 8 & 0\\
+#           -6 & 0 & 0 & 0
+#         \end{bmatrix}.
+#
+# We can solve for the ground state energy using
+# the variational quantum eigensolver (VQE) algorithm.
+#
+# Let's use the ``default.qubit`` simulator for both the analytic gradient,
+# as well as the estimated gradient using number of shots :math:`N=1, 100`.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+np.random.seed(2)
+
+from pennylane import expval
+from pennylane.init import strong_ent_layers_uniform
+from pennylane.templates.layers import StronglyEntanglingLayers
+
+num_layers = 2
+num_wires = 2
+eta = 0.01
+steps = 200
+
+dev_GD = qml.device("default.qubit", wires=num_wires, analytic=True)
+dev_SGD1 = qml.device("default.qubit", wires=num_wires, analytic=False, shots=1)
+dev_SGD100 = qml.device("default.qubit", wires=num_wires, analytic=False, shots=100)
+
+##############################################################################
+# We can use ``qml.Hermitian`` to directly specify we want to measure
+# the expectation value of the dense matrix :math:`H`:
+
+H = np.array([[8, 4, 0, -6],
+              [4, 0, 4, 0],
+              [0, 4, 8, 0],
+              [-6, 0, 0, 0]])
+
+def circuit(params):
+    StronglyEntanglingLayers(*params, wires=[0, 1])
+    return expval(qml.Hermitian(H, wires=[0, 1]))
+
+
+##############################################################################
+# Now, we create three QNodes, each corresponding to a device above,

```

Comment:
This feels clunky. I'm not 100% happy with describing 1 shot vs 100 shots as two different "devices" :frowning: (I think describing as 3 separate QNodes is ok)
 
Maybe we can make it clear that they are all same device in naming scheme somehow,
e.g., 
`qubit_GD_exact`
`qubit_SGD1`
`qubit_SGD100`

Reply:
I realised we can make use of @johannesjmeyer's shots PR, and just create two devices; one analytic, one not. Then we can use the new setter `dev.shots = 1` to change shots when you need it.

Reply:
:+1: